src/Microsoft.VisualStudio.SDK.Analyzers/Types.cs

```diff
@@ -340,6 +340,11 @@ internal static class ThreadHelper
             /// Gets the <see cref="Microsoft.CodeAnalysis.CSharp.Syntax.TypeSyntax"/> for this type.
             /// </summary>
             internal static TypeSyntax TypeSyntax { get; } = Utils.QualifyName(Namespace, SyntaxFactory.IdentifierName(TypeName));
+
+            /// <summary>
+            /// Gets the fully-qualified name of this type as a string.
+            /// </summary>
+            internal static string FullName => string.Join(".", Namespace) + "." + TypeName;
```

Comment:
That's a great point. I copied the example of all the other FullName properties in the various classes in Types.cs. Looks like there are 14 of them (including the one I added) all implemented as lambda expressions. (They probably don't show up in the diff.) I'm happy to update all of them -- should I go ahead with that, or just modify the one I added?

Reply:
If it's easy to do, please do. Thanks for calling out the broader opportunity.

Reply:
Cool, all done. This change has the effect of initializing the FullName property string when the static class is first used, even if the client doesn't care about FullName. (Same applies to the TypeSyntax property.)

Previously with the lambda expression the property value was created upon first invocation, but then thrown away and re-created each time as you noted.

But I guess if the properties are used widely and repeatedly by the analyzer then it may not matter if they're all initialized when the static class is first used.